File: Back/.Net/Modules/Bets/BetFriends.Bets.Infrastructure/Repositories/Sql/DataAccess/BetEntity.cs
using BetFriends.Bets.Domain.Bets;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BetFriends.Bets.Infrastructure.Repositories.Sql.DataAccess;

[Table("bets", Schema = "bet")]
public class BetEntity
{
    public BetEntity()
    {

    }

    public BetEntity(Bet bet)
    {
        Id = bet.BetId.Value;
        Description = bet.Description;
        Coins = bet.Coins;
        EndDate = bet.EndDate;
        Guests = string.Join(';', bet.Guests);
        BettorId = bet.BookieId.Value;
        MaxAnswerDate = bet.MaxAnswerDate.Value;
    }

    [Column("surrogate_id", TypeName = "int")]
    public int SurrogateId { get; set; }
    [Column("id"), Key, Required]
    public Guid Id { get; set; }
    [Column("description", TypeName = "varchar(MAX)"), Required]
    public string Description { get; set; }
    [Column("coins", TypeName = "int"), Required]
    public int Coins { get; set; }
    [Column("end_date", TypeName = "datetime"), Required]
    public DateTime EndDate { get; set; }
    [Column("guests", TypeName = "varchar(MAX)"), Required]
    public string Guests { get; set; }
    [Column("bettor_id"), Required]
    public Guid BettorId { get; set; }
    [Column("max_answer_date", TypeName = "datetime"), Required]
    public DateTime MaxAnswerDate { get; set; }
    [Column("is_successful", TypeName = "bit")]
    public bool? IsSuccessful { get; set; }
}

File: Back/NodeJS/modules/bets/src/infrastructure/DataAccess/entities/BetEntity.ts
import { Column, Entity } from "typeorm";
import { Bet } from "../../../domain/bets/Bet";

@Entity({
    schema: "bet",
    name: "bets"
})
export class BetEntity {
    
    @Column({
        name: "ownerid"
    })
    BetId: string;
    @Column({
        name: "bookie_id"
    })
    BettorId: string;
    @Column({
        name: "coins"
    })
    Coins: number;
    @Column({
        name: "description"
    })
    Description: string;
    @Column({
        name: "end_date"
    })
    EndDate: Date;
    @Column({
        name: "max_answer_date"
    })
    MaxAnswerDate: Date;
    @Column({
        name: "guests"
    })
    Guests: string;
    @Column({
        name: "is_successful"
    })
    IsSuccessful?: boolean;

    static Create(bet: Bet) : BetEntity {
        const entity = new BetEntity();
        entity.BetId = bet.BetId.Value;
        entity.BettorId = bet.BettorId.Value;
        entity.Coins = bet.Coins;
        entity.Description = bet.Description;
        entity.EndDate = bet.EndDate;
        entity.Guests = bet.Members.join(";")
        entity.MaxAnswerDate = bet.MaxAnswerDate;
        entity.IsSuccessful = bet.IsSuccessful;
        return entity;
    }

}